feat: drop WITH(KEY) syntax (#5363)
confluentinc/ksql: docs/developer-guide/joins/partition-data.md

Streams, on the other hand, may not have a defined key or may have a key that
differs from the join column. In these cases, ksqlDB internally repartitions
the stream, which implicitly defines the correct key for it.

!!! important

>     Be aware that Kafka guarantees the relative order of any two messages from

Reviewer: ```suggestion
    {{ site.ak }} guarantees the relative order of any two messages from
```

>     one source partition only if they are also both in the same partition

Reviewer: ```suggestion
    one source partition only if they are both in the same partition
```

>     after the repartition. Otherwise, Kafka is likely to interleave messages.

Reviewer: ```suggestion
    after the repartition. Otherwise, {{ site.ak }} is likely to interleave messages.
```
